app/src/main/java/com/example/parkingsystemjava/mvp/model/ReservationModel.java

```diff
@@ -4,25 +4,32 @@
 import com.example.parkingsystemjava.database.ReservationDataBase;
 import com.example.parkingsystemjava.mvp.contract.ReservationActivityContract;
 import java.util.Calendar;
+import java.util.List;
 
 public class ReservationModel implements ReservationActivityContract.ModelContract {
     private final Reservation reservation = new Reservation();
     private final ReservationDataBase database;
+    private boolean overlap = false;
 
     public ReservationModel(ReservationDataBase database) {
         this.database = database;
     }
 
     @Override
-    public void saveReservation(String parkingLots, String userPassword) {
-        this.reservation.setParkingLots(parkingLots);
-        this.reservation.setUserPassword(userPassword);
-        database.addValues(reservation);
+    public void saveReservation() {
+        if (!existsOverlap(reservation)) {
+            database.addValues(reservation);
+        }
     }
 
-    @Override
-    public Reservation getReservation(String parkingLots, String pass) {
-        return this.database.getReservation(parkingLots, pass);
+    public Reservation createReservation(String parkingLots, String userPassword) {
+        if (reservation.getStartDate() != null && reservation.getEndDate() != null &&
+                !parkingLots.equals("") && !userPassword.equals("")) {
```

Comment:
no hardcode pls, use a EMPTY_STRING constant in the constantUtils class

Reply:
Done